src/main/java/com/minecolonies/core/colony/buildings/AbstractBuildingContainer.java

```diff
@@ -169,14 +169,15 @@ public void registerBlockPosition(@NotNull final Block block, @NotNull final Blo
                 }
             }
         }
-        else if (block instanceof BlockMinecoloniesRack)
+        else
         {
-            addContainerPosition(pos);
-            final BlockEntity entity = world.getBlockEntity(pos);
-            if (entity instanceof TileEntityRack rackEntity)
-            {
-                rackEntity.setBuildingPos(this.getID());
+            BlockEntity entity = world.getBlockEntity(pos);
+            Optional<IStorageBlockInterface> storageInterface = ModStorageBlocks.getStorageBlockInterface(entity);
+            if (storageInterface.isEmpty()) {
+                return;
             }
+            addContainerPosition(pos);
```

Comment:
I'd say the building needs to contain a list of storage blocks itself, not a list of positions.

Reply:
Done.